appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1285,55 +1266,29 @@ else if (requestTracing[0].matches("\\D+")) {
-                        } catch (NumberFormatException e) {
-                            LOGGER.log(Level.WARNING, "{0} is not a valid value for --requestTracingThresholdValue",
-                                    value);
-                            throw e;
-                        }
+                        requestTracingThresholdValue = parseArgument(value, "value for --requestTracingThresholdValue", 
+                                Long::parseLong).longValue();
                         break;
                     case enablerequesttracingadaptivesampling:
                         enableRequestTracingAdaptiveSampling = true;
                         break;
                     case requesttracingadaptivesamplingtargetcount:
                         enableRequestTracingAdaptiveSampling = true;
-                        try {
-                            requestTracingAdaptiveSamplingTargetCount = Integer.parseInt(value);
-                        } catch (NumberFormatException e) {
-                            LOGGER.log(Level.WARNING, "{0} is not a valid value for --requestTracingAdaptiveSamplingTargetCount", value);
-                            throw e;
-                        }
+                        requestTracingAdaptiveSamplingTargetCount = parseArgument(value,
+                                "value for --requestTracingAdaptiveSamplingTargetCount", Integer::parseInt).intValue();
                         break;
                     case requesttracingadaptivesamplingtimevalue:
                         enableRequestTracingAdaptiveSampling = true;
-                        try {
-                            requestTracingAdaptiveSamplingTimeValue = Integer.parseInt(value);
-                        } catch (NumberFormatException e) {
-                            LOGGER.log(Level.WARNING, "{0} is not a valid value for --requestTracingAdaptiveSamplingTimeValue", value);
-                            throw e;
-                        }
+                        requestTracingAdaptiveSamplingTimeValue = parseArgument(value,
+                                "value for --requestTracingAdaptiveSamplingTimeValue", Integer::parseInt).intValue();
                         break;
                     case requesttracingadaptivesamplingtimeunit:
                         enableRequestTracingAdaptiveSampling = true;
-                        try {
-                            TimeUnit.valueOf(value.toUpperCase());
-                            requestTracingAdaptiveSamplingTimeUnit = value.toUpperCase();
-                        } catch (IllegalArgumentException e) {
-                            LOGGER.log(Level.WARNING, "{0} is not a valid value for --requestTracingAdaptiveSamplingTimeUnit", value);
-                            throw e;
-                        }
+                        requestTracingAdaptiveSamplingTimeUnit = parseTimeUnit(value,
+                                "value for --requestTracingAdaptiveSamplingTimeUnit").name();
```

Comment:
Note that this is a slight change in behaviour since the `parseTimeUnit` helper will also allow for the short forms (like `ns`) which originally was not done for this particular variable (`requestTracingAdaptiveSamplingTimeUnit`) while other variable holding `TimeUnit` names did this.

Reply:
Doc PR: https://github.com/payara/Payara-Server-Documentation/pull/479